codespell_lib/_codespell.py

```diff
@@ -423,8 +423,21 @@ def parse_options(args):
         with open(toml_file, 'rb') as f:
             data = tomli.load(f).get('tool', {})
         config.read_dict(data)
-    config.read(cfg_files)
 
+    # Report which config files are going to be used
+    used_cfg_files = []
+    for cfg_file in cfg_files:
+        _cfg = configparser.ConfigParser()
+        _cfg.read(cfg_file)
+        if _cfg.has_section('codespell'):
+            used_cfg_files.append(cfg_file)
+    if len(used_cfg_files) > 0:
+        print('Used config files:\n')
+    for ifile, cfg_file in enumerate(used_cfg_files):
+        print('    %i : %s' % (ifile, cfg_file))
+
+    # Use config files
+    config.read(cfg_files)
```

Comment:
If you want documented/explicit order of priority for these config files, you' need to call `configparser.ConfigParser.read` sequentially on the config files, then merge the resulting configurations manually.

Don't get me wrong, I have nothing against using the undocumented/implicit priorities enforced by `configparser.ConfigParser.read`. But while can rely on these undocumented/implicit priorities to obtain the same undocumented/implicit priorities in codespell, we cannot document them in codspell if they are not documented in `configparser.ConfigParser.read`.

Reply:
I'm convinced by https://github.com/python/cpython/issues/96765#issuecomment-1288117030. :+1: